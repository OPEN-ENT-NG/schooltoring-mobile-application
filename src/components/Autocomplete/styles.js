import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  input: {
    backgroundColor: COLORS.WHITE
  },
  list: {
    backgroundColor: COLORS.WHITE,
    zIndex: 10,
    position: "relative"
  },
  item: {
    padding: 20
  }
});
