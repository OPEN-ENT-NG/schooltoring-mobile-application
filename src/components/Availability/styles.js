import { StyleSheet } from "react-native";

export default StyleSheet.create({
  container: {
    paddingHorizontal: 30,
    paddingVertical: 15,
    flex: 1,
    justifyContent: "space-evenly"
  },
  item: {
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center",
    height: 60
  },
  itemSwitch: {
    marginRight: 20
  },
  itemText: {
    fontSize: 18
  },
  list: {
    flex: 1,
    justifyContent: "space-around"
  },
  title: {
    fontSize: 18,
    marginBottom: 40,
    flex: 0
  },
  buttonView: {
    flexDirection: "row",
    justifyContent: "flex-end",
    flex: 0,
    padding: 15
  }
});
