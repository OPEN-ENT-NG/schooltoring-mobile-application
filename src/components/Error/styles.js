import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column"
  },
  imageCritical: {
    width: 280,
    height: 301,
    position: "absolute",
    bottom: 0,
    left: -30
  },
  imageUncritical: {
    width: 262,
    height: 301,
    position: "absolute",
    bottom: 0,
    right: -30
  }
});

export default styles;
