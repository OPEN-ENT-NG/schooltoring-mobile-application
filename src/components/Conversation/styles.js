import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

const textColor = {
  color: COLORS.TEXT
};

export default StyleSheet.create({
  conversation: {
    flexDirection: "row",
    paddingHorizontal: 10
  },
  avatar: {
    alignSelf: "flex-start",
    flex: 1,
    alignItems: "center",
    paddingVertical: 20,
    marginHorizontal: 5
  },
  content: {
    flex: 4,
    justifyContent: "space-evenly",
    borderBottomColor: "#000",
    borderBottomWidth: StyleSheet.hairlineWidth
  },
  name: {
    ...textColor,
    fontWeight: "bold",
    fontSize: 20
  },
  message: {
    flexDirection: "row",
    justifyContent: "space-evenly"
  },
  icon: {
    fontSize: 20,
    marginRight: 10
  },
  text: {
    ...textColor,
    flex: 5,
    fontSize: 16,
    flexWrap: "wrap"
  },
  date: {
    flexGrow: 2,
    alignItems: "flex-end"
  },
  dateText: {
    ...textColor
  }
});
