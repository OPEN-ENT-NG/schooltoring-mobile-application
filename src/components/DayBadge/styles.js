import { StyleSheet } from "react-native";

import { COLORS } from "../../styles/common";

export default StyleSheet.create({
  item: {
    justifyContent: "center",
    alignItems: "center",
    width: 30,
    height: 30,
    backgroundColor: COLORS.GREY
  },
  itemText: {
    flex: 1,
    color: COLORS.WHITE,
    fontSize: 20
  },
  available: {
    backgroundColor: COLORS.NEGATIVE
  }
});
