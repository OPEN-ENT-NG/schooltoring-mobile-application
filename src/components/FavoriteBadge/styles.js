import { StyleSheet, Platform } from "react-native";

import { COLORS } from "../../styles/common";

const shadow =
  Platform.OS === "android"
    ? {
        elevation: 4
      }
    : {
        shadowColor: COLORS.GREY,
        shadowOffset: { width: 2, height: 2 },
        shadowOpacity: 0.6,
        shadowRadius: 1
      };

export default StyleSheet.create({
  badge: {
    alignItems: "stretch",
    width: "50%",
    padding: 10
  },
  top: {
    aspectRatio: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: COLORS.SECONDARY_BACKGROUND,
    ...shadow
  },
  bottom: {
    backgroundColor: COLORS.BACKGROUND,
    ...shadow,
    maxHeight: 120
  },
  name: {
    textAlign: "center",
    color: COLORS.TEXT,
    fontWeight: "bold",
    fontSize: 16
  },
  buttons: {
    flexDirection: "row",
    justifyContent: "space-around",
    alignItems: "center",
    padding: 5
  },
  chatButton: {
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: COLORS.PRIMARY
  },
  loader: {
    ...StyleSheet.absoluteFillObject,
    backgroundColor: "rgba(255, 255, 255, 0.8)",
    elevation: 1000,
    zIndex: 5
  }
});
