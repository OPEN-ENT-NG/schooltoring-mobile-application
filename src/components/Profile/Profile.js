import React from "react";
import { View, Text, ScrollView } from "react-native";
import Icon from "react-native-vector-icons/MaterialIcons";
import PropTypes from "prop-types";

import I18n from "../../api/I18n";

import { Avatar } from "../Avatar/Avatar";
import { SecondaryButton } from "../SecondaryButton/SecondaryButton";
import { SubjectBadge } from "../SubjectBadge/SubjectBadge";
import Touchable from "../Touchable/Touchable";

import { COLORS } from "../../styles/common";
import styles from "./styles";

export const Profile = props => {
  getSubjectsName = data => {
    return props.subjects
      .filter(({ subjectId }) =>
        data.find(({ subject_id }) => subject_id === subjectId)
      )
      .map(({ subjectLabel }) => subjectLabel);
  };

  renderSubject = (item, backgroundColor) => {
    return (
      <SubjectBadge
        key={item}
        style={{
          backgroundColor: backgroundColor
        }}
        title={item}
      />
    );
  };

  renderAvailability = item => {
    return (
      <View
        key={item}
        style={[
          styles.item,
          {
            backgroundColor:
              props.profile.availabilities[item] === true
                ? COLORS.NEGATIVE
                : COLORS.GREY
          }
        ]}
      >
        <Text style={styles.itemText}>
          {I18n.t(`availability.${item}`)
            .charAt(0)
            .toUpperCase()}
        </Text>
      </View>
    );
  };

  const ProfileItem = props => {
    return (
      <View id={props.id}>
        <View style={styles.title}>
          <Text style={styles.titleText}>{props.title}</Text>
          <Touchable onPress={() => props.edit()}>
            <Icon style={styles.titleIcon} name="create" />
          </Touchable>
        </View>

        <View style={[styles.list, props.listStyle]}>
          {props.data.map(item => props.renderItem(item))}
        </View>
      </View>
    );
  };

  return (
    <View style={styles.flex}>
      <ScrollView contentContainerStyle={styles.container}>
        <View style={styles.top}>
          <Avatar
            size={120}
            src={`${global.config.auth.endpoint}${props.userinfo.avatar}`}
          />

          <View style={styles.topRight}>
            <Text style={styles.name} id="username-field">
              {props.userinfo.username}
            </Text>
            <Text style={styles.titleText} id="classname-field">
              {props.userinfo.classNames[0].split("$")[1]}
            </Text>
            {props.showTopButtons && (
              <View style={styles.topRightButtons}>
                <Icon style={styles.titleIcon} name="stars" size={30} />
                <Text style={[styles.titleText, { marginRight: 20 }]}>
                  {I18n.t("badges")}
                </Text>
                <Icon style={styles.titleIcon} name="favorite" size={30} />
                <Text style={styles.titleText}>{I18n.t("favorites")}</Text>
              </View>
            )}
          </View>
        </View>

          <View style={styles.flex}>
            <ProfileItem
              id="strengths-section"
              title={I18n.t("strength.title")}
              edit={() => props.navigation.push("Strength")}
              data={getSubjectsName(props.profile.strengths)}
              renderItem={item => renderSubject(item, COLORS.PRIMARY)}
            />

          <View style={styles.divider} />

            <ProfileItem
              id="weaknesses-section"
              title={I18n.t("weakness.title")}
              edit={() => props.navigation.push("Weakness")}
              data={getSubjectsName(props.profile.weaknesses)}
              renderItem={item => renderSubject(item, COLORS.SECONDARY)}
            />

          <View style={styles.divider} />

          <ProfileItem
            id="availabilities-section"
            title={I18n.t("availability.title")}
            edit={() => props.navigation.push("Availability")}
            data={Object.keys(props.profile.availabilities)}
            listStyle={{ justifyContent: "space-evenly" }}
            renderItem={item => renderAvailability(item)}
          />
        </View>
      </ScrollView>
      {!props.showTopButtons && (
        <View style={styles.buttonView}>
          <SecondaryButton
            style={styles.button}
            onPress={() => props.saveProfile()}
            title={I18n.t("save")}
          />
        </View>
      )}
    </View>
  );
};

Profile.propTypes = {
  subjects: PropTypes.arrayOf(
    PropTypes.shape({
      subjectId: PropTypes.string,
      subjectCode: PropTypes.string,
      subjectLabel: PropTypes.string
    })
  ).isRequired,
  profile: PropTypes.shape({
    availabilities: PropTypes.shape({
      monday: PropTypes.bool.isRequired,
      tuesday: PropTypes.bool.isRequired,
      wednesday: PropTypes.bool.isRequired,
      thursday: PropTypes.bool.isRequired,
      friday: PropTypes.bool.isRequired,
      saturday: PropTypes.bool.isRequired,
      sunday: PropTypes.bool.isRequired
    }),
    weaknesses: PropTypes.arrayOf(
      PropTypes.shape({ subject_id: PropTypes.string.isRequired })
    ),
    strengths: PropTypes.arrayOf(
      PropTypes.shape({ subject_id: PropTypes.string.isRequired })
    )
  }).isRequired,
  userinfo: PropTypes.object.isRequired,
  navigation: PropTypes.object.isRequired,
  showTopButtons: PropTypes.bool,
  saveProfile: PropTypes.func
};
