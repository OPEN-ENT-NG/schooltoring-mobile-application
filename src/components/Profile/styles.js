import { StyleSheet, Dimensions } from "react-native";
import { COLORS } from "../../styles/common";

const text = { color: COLORS.TEXT };

export default StyleSheet.create({
  flex: {
    flex: 1
  },
  container: {
    paddingHorizontal: 30,
    paddingVertical: 15,
    flexGrow: 1,
    justifyContent: "space-evenly"
  },
  divider: {
    borderBottomColor: COLORS.LIGHT_GREY,
    borderBottomWidth: 1,
    marginVertical: 20
  },
  topRightButtons: {
    flexDirection: "row",
    justifyContent: "space-evenly",
    marginBottom: 15
  },
  topRightButton: {
    flexDirection: "row",
    alignItems: "center",
    padding: 15,
    width: "100%"
  },
  title: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: 10
  },
  titleIcon: {
    ...text,
    fontSize: 25,
    marginRight: 5,
    marginLeft: 5
  },
  titleText: {
    ...text,
    fontSize: 18
  },
  list: {
    flexGrow: 1,
    flexDirection: "row",
    justifyContent: "flex-start",
    flexWrap: "wrap"
  },
  buttonView: {
    flexDirection: "row",
    justifyContent: "flex-end",
    flex: 0,
    padding: 15
  }
});
