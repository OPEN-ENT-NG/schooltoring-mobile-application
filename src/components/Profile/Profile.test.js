import React from "react";
import { Text } from "react-native";
import { shallow } from "enzyme";

import { Profile } from "./Profile";
import Touchable from "../Touchable/Touchable";

import state from "../../jest/state.json";
import { COLORS } from "../../styles/common";

const navigation = { push: jest.fn() };
const strengthsName = [
  "SCIENCES PHYSIQUES ET CHIMIE APPLIQUEE",
  "SCIENCES INDUSTRIELLES DE L'INGENIEUR",
  "MATHEMATIQUES"
];
const weaknessesName = [
  "ESPAGNOL LV2",
  "HISTOIRE-GEOGRAPHIE",
  "ECONOMIE-DROIT"
];
const availabilities = [false, true, false, true, false, false, true];

beforeAll(() => {
  global.config = { auth: { endpoint: "http://ent3-core.gdapublic.fr" } };
});

describe("Render", () => {
  let rendered;

  test("Profile should render without crashing", () => {
    rendered = shallow(
      <Profile {...state} navigation={navigation} saveProfile={jest.fn()} />
    );
    expect(rendered).toMatchSnapshot();
  });

  test(`Username and Class name should be respectively '${
    state.userinfo.username
  }' and '${state.userinfo.classNames[0].split("$")[1]}'`, () => {
    const usernameText = rendered.find(Text).find('[id="username-field"]');
    const classnameText = rendered.find(Text).find('[id="classname-field"]');

    expect(usernameText.contains(state.userinfo.username)).toBeTruthy();
    expect(
      classnameText.contains(state.userinfo.classNames[0].split("$")[1])
    ).toBeTruthy();
  });

  test("Profile render strengths should be the same as props", () => {
    const badges = rendered
      .find('[id="strengths-section"]')
      .dive()
      .find("SubjectBadge");

    expect(badges.length).toEqual(strengthsName.length);
    for (let i = 0; i < strengthsName.length; i++) {
      expect(
        badges
          .at(i)
          .dive()
          .find(Touchable)
          .contains(strengthsName[i])
      ).toBeTruthy();
    }
  });

  test("Profile render weaknesses should be the same as props", () => {
    const badges = rendered
      .find('[id="weaknesses-section"]')
      .dive()
      .find("SubjectBadge");

    expect(badges.length).toEqual(weaknessesName.length);
    for (let i = 0; i < weaknessesName.length; i++) {
      expect(
        badges
          .at(i)
          .dive()
          .find(Touchable)
          .contains(weaknessesName[i])
      ).toBeTruthy();
    }
  });

  test("Profile render availabilities should be the same as props", () => {
    const days = rendered
      .find('[id="availabilities-section"]')
      .dive()
      .childAt(1)
      .children();

    expect(days.length).toEqual(availabilities.length);
    for (let i = 0; i < days.length; i++) {
      let day = days.get(i);
      expect(day.props.style[1].backgroundColor).toEqual(
        availabilities[i] ? COLORS.NEGATIVE : COLORS.GREY
      );
    }
  });
});

describe("Interactions", () => {
  test("ProfileItem editing button should be called on press", () => {
    const rendered = shallow(
      <Profile {...state} navigation={navigation} saveProfile={jest.fn()} />
    );

    const button = rendered
      .find('[id="strengths-section"]')
      .dive()
      .find(Touchable);

    button.prop("onPress")();

    expect(navigation.push).toHaveBeenCalled();
  });
});
