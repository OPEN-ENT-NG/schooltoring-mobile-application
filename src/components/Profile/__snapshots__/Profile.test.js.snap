// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Profile should render without crashing 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "justifyContent": "space-evenly",
        "paddingHorizontal": 30,
        "paddingVertical": 15,
      }
    }
  >
    <Component
      style={
        Object {
          "flex": 0,
          "flexDirection": "row",
          "marginBottom": 20,
        }
      }
    >
      <Avatar
        size={120}
        src="http://ent3-core.gdapublic.fr/workspace/document/49a54de3-51f8-464b-81bb-a377f508f382"
      />
      <Component
        style={
          Object {
            "justifyContent": "space-evenly",
            "marginLeft": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          id="username-field"
          style={
            Object {
              "color": "#232323",
              "fontSize": 26,
              "fontWeight": "bold",
            }
          }
        >
          John DOE
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          id="classname-field"
          style={
            Object {
              "color": "#232323",
              "fontSize": 20,
            }
          }
        >
          2DE4
        </Text>
      </Component>
    </Component>
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <ProfileItem
        data={
          Array [
            "SCIENCES PHYSIQUES ET CHIMIE APPLIQUEE",
            "SCIENCES INDUSTRIELLES DE L'INGENIEUR",
            "MATHEMATIQUES",
          ]
        }
        edit={[Function]}
        id="strengths-section"
        renderItem={[Function]}
        title="Mes points forts"
      />
      <Component
        style={
          Object {
            "borderBottomColor": "#ebebeb",
            "borderBottomWidth": 1,
            "marginVertical": 20,
          }
        }
      />
      <ProfileItem
        data={
          Array [
            "ESPAGNOL LV2",
            "HISTOIRE-GEOGRAPHIE",
            "ECONOMIE-DROIT",
          ]
        }
        edit={[Function]}
        id="weaknesses-section"
        renderItem={[Function]}
        title="Mes points faibles"
      />
      <Component
        style={
          Object {
            "borderBottomColor": "#ebebeb",
            "borderBottomWidth": 1,
            "marginVertical": 20,
          }
        }
      />
      <ProfileItem
        data={
          Array [
            "monday",
            "tuesday",
            "wednesday",
            "thursday",
            "friday",
            "saturday",
            "sunday",
          ]
        }
        edit={[Function]}
        id="availabilities-section"
        listStyle={
          Object {
            "justifyContent": "space-evenly",
          }
        }
        renderItem={[Function]}
        title="Mes dispos"
      />
    </Component>
  </ScrollView>
  <Component
    style={
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "padding": 15,
      }
    }
  >
    <SecondaryButton
      onPress={[Function]}
      title="ENREGISTRER"
    />
  </Component>
</Component>
`;
